File: DSA/Day_47_Question_1.py
from collections import deque

# Function to insert element at rear
def insert_rear(queue, element):
    queue.append(element)
    return list(queue)

# Function to insert element at front
def insert_front(queue, element):
    queue.appendleft(element)
    return list(queue)

# Function to find front and rear of the queue
def front_and_rear(queue):
    if not queue:
        return "Queue is empty."
    else:
        return f"Front: {queue[0]}, Rear: {queue[-1]}"

# Example Usage
q1 = deque([1, 2])
print("Insert at Rear:", insert_rear(q1, 3))  # [1, 2, 3]

q2 = deque([2, 3, 4])
print("Insert at Front:", insert_front(q2, 1))  # [1, 2, 3, 4]

q3 = deque([10, 20, 30])
print("Front and Rear:", front_and_rear(q3))  # Front: 10, Rear: 30

q4 = deque([])
print("Front and Rear:", front_and_rear(q4))  # Queue is empty.

File: DSA/Gautam_Shyamnani_590013078/Day47Question1.cpp
#include <iostream>
#include <queue>
using namespace std;

// Function to insert element at the rear of the queue
queue<int> insertAtRear(queue<int> q, int value) {
    q.push(value); // Think of this as someone stepping into the line
    return q;
}

int main() {
    queue<int> q;
    q.push(1);
    q.push(2);

    int newElement = 3;

    // Call the function to insert at rear
    queue<int> updatedQueue = insertAtRear(q, newElement);

    // Show the updated queue
    cout << "Queue after insertion: ";
    while (!updatedQueue.empty()) {
        cout << updatedQueue.front() << " ";
        updatedQueue.pop();
    }

    return 0;
}
